core/gotemplate/engine_test.go

```diff
@@ -0,0 +1,183 @@
+package gotemplate
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"reflect"
+	"testing"
+	"time"
+
+	"flamingo.me/flamingo/v3/framework/flamingo"
+)
+
+var noAdditionalTemplateFuncs = func() map[string]flamingo.TemplateFunc {
+	return make(map[string]flamingo.TemplateFunc)
+}
+
+var additionalTemplateFuncs = func() map[string]flamingo.TemplateFunc {
+	funcs := make(map[string]flamingo.TemplateFunc)
+	funcs["customTemplateFunc"] = CustomTemplateFunc{}
+	return funcs
+}
+
+type CustomTemplateFunc struct{}
+
+var _ flamingo.TemplateFunc = CustomTemplateFunc{}
+
+func (CustomTemplateFunc) Func(ctx context.Context) interface{} {
+	return func() interface{} {
+		return "test-abc"
+	}
+}
+
+func Test_engine_Render(t *testing.T) {
+	type fields struct {
+		templatesBasePath  string
+		layoutTemplatesDir string
+		debug              bool
+		tplFuncs           func() map[string]flamingo.TemplateFunc
+	}
+	type args struct {
+		name string
+		data interface{}
+	}
+	tests := []struct {
+		name    string
+		fields  fields
```

Comment:
It is not clear what `fields` and `args` mean, maybe rename fields to engineConfig and args to renderArgs?